packages/expo-video/src/VideoView.tsx

```diff
@@ -84,3 +90,43 @@ function getPlayerId(player: number | VideoPlayer): number | null {
   }
   return null;
 }
+
+/**
+ * Returns a shared object, which is automatically cleaned up when the component is unmounted.
+ */
+// TODO: when SharedObject type is added make T extend it
+export function useReleasingSharedObject<T>(factory: () => T, dependencies: DependencyList): T {
+  const objectRef = useRef<any | null>(null);
+  const isFastRefresh = useRef(false);
+  const previousDependencies = useRef<DependencyList>(dependencies);
+
+  if (objectRef.current == null) {
+    objectRef.current = factory();
+  }
+
+  const object = useMemo(() => {
+    // If the dependencies have changed, release the previous object and create a new one, otherwise this has been called
+    // because of a fast-refresh and we don't want to release the object.
+    if (!previousDependencies.current.every((value, index) => value === dependencies[index])) {
+      objectRef.current.release();
+      objectRef.current = factory();
+      previousDependencies.current = dependencies;
+    } else {
+      isFastRefresh.current = true;
+    }
+    return objectRef.current;
+  }, dependencies);
+
+  useEffect(() => {
+    isFastRefresh.current = false;
+
+    return () => {
+      // This will be called on every fast-refresh and on unmount, but we only want to release the player on unmount.
```

Comment:
It's no longer a player 🙂 
```suggestion
      // This will be called on every fast refresh and on unmount, but we only want to release the object on unmount.
```